assets/src/edit-story/utils/processPastedNodeList.js

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import escapeHTML from './escapeHTML';
+
+const ALLOWED_CONTENT_NODES = ['strong', 'em', 'u'];
+
+function processNodeContent(node) {
+  const tag = node.parentNode?.tagName?.toLowerCase();
+  const stripTags = !ALLOWED_CONTENT_NODES.includes(tag);
+  const content = escapeHTML(node.textContent);
+  if (stripTags) {
+    return content;
+  }
+  return `<${tag}>${content}<${tag}/>`;
+}
+
+export default function processPastedNodeList(nodeList, content) {
+  for (let i = 0; i < nodeList.length; i++) {
+    const n = nodeList[i];
```

Comment:
If you still want to manually walk the tree, you'll probably make it easier for yourself if you wrap it in `Array.from` here and then use native array functionality like `.map()` and `.reduce()` - and not those old-school loops.

Reply:
This one is just to simplify the code a lot - shouldn't require a huge change.

Reply:
Switched to reducer in ad8a5d5